src/views/nodeCache/Trie.ts

```diff
@@ -0,0 +1,112 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license.
+
+import * as path from "path";
+import { Uri } from "vscode";
+import { DataNode } from "../dataNode";
+
+export class Trie {
+    private root: TrieNode;
+
+    constructor() {
+        this.root = new TrieNode(null, null);
+    }
+
+    public insert(input: DataNode): void {
+        let currentNode: TrieNode = this.root;
+        const fsPath: string = Uri.parse(input.uri).fsPath;
+        const segments: string[] = fsPath.split(path.sep);
+
+        for (const segment of segments) {
```

Comment:
Here looks like you're using fspath as trie key, does it work for invisible project?

Reply:
For invisible project, since the uris of the project node and package fragment root node are the same. So in trie, package fragment root node is stored. (actually it replaces the project node when it gets rendered).

Luckily, the functionality is still working.